### DISCUSSION TITLE
About the transposed convolution

### DISCUSSION THREAD

**Discussion by ilan-theodoro** [Project-MONAI/GenerativeModels#428] [Q&A]
Hi everyone! This project is awesome, thank you all for the initiative! 

I was looking at your VQVAE implementation, and I noticed that you adopt transposed convolution as your upsample method. 

https://github.com/Project-MONAI/GenerativeModels/blob/1b34e60f80555e02036915224dd1783409095d3c/generative/networks/nets/vqvae.py#L257C21-L257C40 

As far as I know, transposed convolutions have the undesired effect of possibly introducing the checkerboard effect. It is still used in the [original VQVAE implementation](https://github.com/MishaLaskin/vqvae/blob/d761a999e2267766400dc646d82d3ac3657771d4/models/decoder.py#L28C16-L28C31), but the authors of VQGAN [chose to adopt interpolation](https://github.com/CompVis/taming-transformers/blob/3ba01b241669f5ade541ce990f7650a3b8f65318/taming/modules/diffusionmodules/model.py#L50C1-L50C44) as their upsampling method. 

Did you only implement the original VQVAE or did you evaluate and choose to use transposed convolution? If the former, a flag to select simple interpolation would be helpful :)

**Comment by marksgraham** [comment]
I recall @danieltudosiu encountering checkerboard artefacts and experimenting with ways to mitigate them - do you remember experimenting with interpolation for upscaling, dan?
